e2e/evm/test.go

```diff
@@ -36,49 +36,67 @@ func SetupEVM2EVEMTestSuite(fabric1, fabric2 calls.TxFabric, endpoint1, endpoint
 
 type IntegrationTestSuite struct {
 	suite.Suite
-	client            TestClient
-	client2           TestClient
-	bridgeAddr        common.Address
-	erc20HandlerAddr  common.Address
-	erc20ContractAddr common.Address
-	fabric1           calls.TxFabric
-	fabric2           calls.TxFabric
-	endpoint1         string
-	endpoint2         string
-	adminKey          *secp256k1.Keypair
+	client             TestClient
+	client2            TestClient
+	gasPricer          calls.GasPricer
+	bridgeAddr         common.Address
+	erc20HandlerAddr   common.Address
+	erc20ContractAddr  common.Address
+	genericHandlerAddr common.Address
+	assetStoreAddr     common.Address
+	fabric1            calls.TxFabric
+	fabric2            calls.TxFabric
+	endpoint1          string
+	endpoint2          string
+	adminKey           *secp256k1.Keypair
+	erc20RID           [32]byte
+	genericRID         [32]byte
 }
 
-func (s *IntegrationTestSuite) SetupSuite()    {}
-func (s *IntegrationTestSuite) TearDownSuite() {}
-func (s *IntegrationTestSuite) SetupTest() {
+func (s *IntegrationTestSuite) SetupSuite() {
 	ethClient, err := evmclient.NewEVMClientFromParams(s.endpoint1, s.adminKey.PrivateKey())
 	if err != nil {
 		panic(err)
 	}
+	s.client = ethClient
+
 	ethClient2, err := evmclient.NewEVMClientFromParams(s.endpoint2, s.adminKey.PrivateKey())
 	if err != nil {
 		panic(err)
 	}
+	s.client2 = ethClient2
+
 	b, err := ethClient.LatestBlock()
 	if err != nil {
 		panic(err)
 	}
+
 	log.Debug().Msgf("Latest block %s", b.String())
-	bridgeAddr, erc20Addr, erc20HandlerAddr, err := PrepareEVME2EEnv(ethClient, s.fabric1, 1, big.NewInt(1), s.adminKey.CommonAddress(), big.NewInt(0))
+
+	config, err := PrepareEVME2EEnv(ethClient, s.fabric1, 1, big.NewInt(1), s.adminKey.CommonAddress())
 	if err != nil {
 		panic(err)
 	}
-	s.client = ethClient
-	s.client2 = ethClient2
-	s.bridgeAddr = bridgeAddr
-	s.erc20ContractAddr = erc20Addr
-	s.erc20HandlerAddr = erc20HandlerAddr
-	//Contract addresses should be the same
-	_, _, _, err = PrepareEVME2EEnv(ethClient2, s.fabric2, 2, big.NewInt(1), s.adminKey.CommonAddress(), big.NewInt(0))
+
+	s.bridgeAddr = config.bridgeAddr
+	s.erc20ContractAddr = config.erc20Addr
+	s.erc20HandlerAddr = config.erc20HandlerAddr
+	s.genericHandlerAddr = config.genericHandlerAddr
+	s.assetStoreAddr = config.assetStoreAddr
+
+	_, err = PrepareEVME2EEnv(ethClient2, s.fabric2, 2, big.NewInt(1), s.adminKey.CommonAddress())
```

Comment:
`PrepareEVME2EEnv` accepts a `uint8` for `domainID` 

Should this `2` need to be wrapped in a `uint8()`?